### PR TITLE
Fogbow dependencies removal

### LINKED ISSUE #61: remove fogbow dependencies from dispatcher component
Although these fogbow deps are standalone (that is, they do not require to deploy any service), I think it is better to be simpler (I'm sure it would be simpler to have some scripts in the saps to do the same)

### PR DESCRIPTION
**Pull request by wesleymonte**
#61. Removed a Fogbow dependency related to the Keystone Identity Service from Saps.
The next step is moves the use of Keystone Identity Service to Swift Permanent Storage.

### CODE REVIEW

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        throws KeystoneException {
+        String[] credentialsSet = {
+            AUTH_URL,
+            PROJECT_ID,
+            PASSWORD,
+            USER_ID
+        };
+        for (String credential : credentialsSet) {
+            String value = credentials.get(credential);
+            if(Objects.isNull(value)) {
```
Comment (thiagoyeds) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java]: Add another condition to not be empty

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneException.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.fogbowcloud.saps.engine.core.dispatcher.email.keystone;
+
+import java.io.IOException;
```
Comment (thiagoyeds) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneException.java]: Remove it

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import java.util.Objects;
+import org.apache.commons.codec.Charsets;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.log4j.Logger;
+import org.eclipse.jetty.http.HttpStatus;
+import org.fogbowcloud.saps.engine.utils.SapsPropertiesUtil;
```
Comment (thiagoyeds) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java]: Remove it

**Review by thiagoyeds — CHANGES_REQUESTED** [review]
@wesleymonte this PR is almost ready to be accepted, however as already said by @thiagomanel do not use the term `folder`, change them all to `dir` and solve the requests.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java (hunk 1 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java]: Done.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneException.java (hunk 2 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneException.java]: Done.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java (hunk 3 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java]: Done.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java (hunk 1 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java]: It's ok

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java (hunk 3 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityPlugin.java]: It's ok

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneException.java (hunk 2 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneException.java]: @wesleymonte 
There was confusion when I asked to remove this, I was referring to the import that was not being used:
```java
import java.io.IOException;
```
But you removed the class and used IOException/Exception, please return the class and remove only that dead import. I will add this order in the next review.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityRequestHelper.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    private static final String IDENTITY_PROP = "identity";
+    private static final String PROJECT_PROP = "project";
+    private static final String METHODS_PROP = "methods";
+    private static final String SCOPE_PROP = "scope";
+    private static final String AUTH_PROP = "auth";
+    private static final String USER_PROP = "user";
+    private static final String ID_PROP = "id";
+    private static final String V3_TOKENS_ENDPOINT_PATH = "/v3/auth/tokens";
+
+    public static String createAccessId(Map<String, String> credentials) throws Exception {
```
Comment (thiagoyeds) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityRequestHelper.java]: As commented [here](https://github.com/ufcg-lsd/saps-engine/pull/88#discussion_r386968642), return KeystoneException class and use it again.

**Review by thiagoyeds — CHANGES_REQUESTED** [review]
@wesleymonte, I have just one more request and it's OK.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityRequestHelper.java (hunk 4 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityRequestHelper.java]: I replaced the Keystone Exception with more specific exceptions like JSONException and IOException.

**Review by thiagoyeds — APPROVED** [review]
lgtm, @thiagomanel review this PR now.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/ProcessedImagesEmailBuilder.java [hunk 5]
```diff
@@ -110,6 +103,8 @@ private JSONArray generateAllTasksJsons(StringBuilder errorBuilder) {
 				} catch (JSONException e1) {
 					LOGGER.error("Failed to create UNAVAILABLE task json.", e);
 				}
+			} catch (Exception e) {
+				e.printStackTrace();
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/ProcessedImagesEmailBuilder.java]: printStackTrace does not help us debug. Should we log instead?

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneException.java (hunk 2 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneException.java]: I saw that it was solved in another way that seems ideal, it's ok

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/ProcessedImagesEmailBuilder.java (hunk 5 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/ProcessedImagesEmailBuilder.java]: I updated exception catchers of this method.

**Review by thiagoyeds — APPROVED** [review]
lgtm, @thiagomanel review this now

**Review by thiagomanel — APPROVED** [review]
lgtm

**Pull request merged**